src/validation.cpp

```diff
@@ -5399,6 +5399,11 @@ bool ChainstateManager::PopulateAndValidateSnapshot(
                       coins_count - coins_left);
             return false;
         }
+        if (coin.out.nValue < 0) {
```

Comment:
While we're at it, might as well check that we're bounded by MAX_MONEY too (as in tx_check: https://github.com/bitcoin/bitcoin/blob/master/src/consensus/tx_check.cpp#L26-L29)

Reply:
Right, I should have just used `MoneyRange` right away...

Reply:
done